### PR TITLE
Add contributing guide

### LINKED ISSUE #1: Need contributing guidelines
# Background

This repo was designed to host the contributing guidelines. 

# Problem

There are no contributing guidelines currently present. 

# Solution

Add initial contributing guidelines.

### PR DESCRIPTION
**Pull request by biohazel**
Addresses #1.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+Anyone may participate in peer review which is expressed by comments in the pull
+request. Typically reviewers will review the code for obvious errors, as well as
+test out the patch set and opine on the technical merits of the patch. PR should
+be reviewed first on the conceptual level before focusing on code style or
+grammar fixes.
+
+
+References & Going Further
+-------------
+These contributing guidelines were based on the [`lightningdevkit/rust-lightning/CONTRIBUTING.md`](https://github.com/lightningdevkit/rust-lightning/blob/main/CONTRIBUTING.md) guide, it may be beneficial to explore this guide further to see how other projects operate.
```
Comment (rrybarczyk) [inline CONTRIBUTING.md:97]: This is still referencing LDK's contributing guide.

**Review by rrybarczyk — CHANGES_REQUESTED** [review]
Just remove the reference of LDK.

File: CONTRIBUTING.md (hunk 1 above)
Comment (biohazel) [reply CONTRIBUTING.md:97]: Actually this is a reference that we want to keep.

**Review by rrybarczyk — APPROVED** [review]
Lgtm!

**Pull request merged**